src/main/scala/learnshapeless/HLists.scala

```diff
@@ -78,20 +88,32 @@ object HLists extends App {
   object IsAustralianPoly extends DefaultIdentityMapping {
     implicit def isAustralia = at[Country](_ == Australia)
   }
-  def ex_poly_country = ???
+
+  def ex_poly_country: String :: Int :: Boolean :: HNil =
+    ex_poly.map(IsAustralianPoly)
 
   def eg_index: Country = eg1_einstein(2)
 
   /* Extract the 3rd element of `ex_poly_country` using a index.
   Does the result surprise you? Why did it happen this way? */
-  def ex_poly_country_element_3rd = ???
-  print(ex_poly_country_element_3rd)
+  def ex_poly_country_element_3rd: Boolean =
+    ex_poly_country(2)
+  println(s"ex_poly_country_element_3rd $ex_poly_country_element_3rd")
 
   def eg_mapped_by_index = eg1_einstein.updateAtWith(2)(_ == Australia)
 
   /* Transform the surname field in `ex_newton` to the first name by looking it up in `Data.scientistsFirstNames`.
   * Use updateAtWith to identify the field by numeric index rather than y type.
   * `updateAtWith` returns the old value, and the updated HList in a tuple */
-  def ex_to_firstname_by_index: (String, HList) = ???
+  def ex_to_firstname_by_index: (String, HList) =
+    ex_newton.updateAtWith(0)(Data.scientistsFirstNames.getOrElse(_, "Unknown"))
+  println(s"ex_to_firstname_by_index $ex_to_firstname_by_index")
+
+  ex_newton match {
+    case matched @ ("Newton" :: year :: country) =>
+      println(s"Matched!!! $matched")
+    case mismatched =>
+      println(s"Failure to match $ex_to_firstname_by_index")
```

Comment:
Perhaps also mention pattern match does work on `HList` - or add a new exercise to do so?